test/suites/drift/suite_test.go

```diff
@@ -337,6 +337,115 @@ var _ = Describe("Drift", Label("AWS"), func() {
 		Entry("DetailedMonitoring Drift", "DetailedMonitoring", v1alpha1.AWSNodeTemplateSpec{DetailedMonitoring: awssdk.Bool(true)}),
 		Entry("AMIFamily Drift", "AMIFamily", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{AMIFamily: awssdk.String(v1alpha1.AMIFamilyBottlerocket)}}),
 	)
+	It("should delete a replacement machine from drift if it fails to register", func() {
```

Comment:
```suggestion
	It("should not continue to drift if a node never registers", func() {
```

Reply:
I think we are mainly validating what we do if the node never registers, whether the node gets cleaned up when it doesn't register is handled by a test inside of the machine test suite